packages/framework/src/Model/Newsletter/NewsletterFacade.php

```diff
@@ -108,4 +109,13 @@ public function deleteSubscribedEmailIfExists(string $email, int $domainId): voi
 
         $this->delete($newsletterSubscriber);
     }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Customer\User\CustomerUser $customerUser
+     * @return bool
+     */
+    public function isSubscribed(CustomerUser $customerUser): bool
+    {
+        return $this->findNewsletterSubscriberByEmailAndDomainId($customerUser->getEmail(), $customerUser->getDomainId()) !== null;
```

Comment:
can we talk about this commit please? 

We're replacing a simple boolean on already hydrated entity with potentialy expensive database query

Reply:
discussed, keeping as is, I will create an issue for optimizing the current customer queries on the SF